packages/yarnpkg-builder/sources/tools/makeConfig.ts

```diff
@@ -34,10 +60,12 @@ export const makeConfig = (config: webpack.Configuration) => merge({
         options: {
           compilerOptions: {
             declaration: false,
-            module: `ESNext`,
-            moduleResolution: `node`,
+            module: `ESNext` as any,
+            moduleResolution: `node` as any,
           },
-        },
+          onlyCompileBundledFiles: true,
+          transpileOnly: true,
+        } as tsLoader.Options,
```

Comment:
Rather than using `as tsLoader.Options`, can we instead do something like:

```ts
const identity = <T>(value: T) => value;

// then later, instead of:
const foo = {...} as tsLoader.Options;
// prefer to do:
const foo = identity<tsLoader.Options>({...});
```

This way we get a stricter typecheck without dangerous cast.

Reply:
Done. 👍